client/components/themenew/hooks/use-theme-showcase-tracks.js

```diff
@@ -0,0 +1,66 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { property, snakeCase } from 'lodash';
+import { useSelector } from 'calypso/state';
+import { DEFAULT_THEME_QUERY } from 'calypso/state/themes/constants';
+import { getThemeType, isThemeActive } from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { useThemeCollectionContext } from '../theme-collection-context';
+import { useThemeContext } from '../theme-context';
+import { useThemeShowcaseContext } from '../theme-showcase-context';
+
+export default function useThemeShowcaseTracks() {
+	const { selectedStyleVariation, themeId, themePosition } = useThemeContext();
+	const { collectionId, collectionPosition } = useThemeCollectionContext();
+
+	const siteId = useSelector( getSelectedSiteId );
+	const isActive = useSelector( ( state ) => isThemeActive( state, themeId, siteId ) );
+	const themeType = useSelector( ( state ) => getThemeType( state, themeId ) );
+
+	const { filterString, query, themes } = useThemeShowcaseContext();
+	const {
+		filter = DEFAULT_THEME_QUERY.filter,
+		number = DEFAULT_THEME_QUERY.number,
+		page = DEFAULT_THEME_QUERY.page,
+		search = DEFAULT_THEME_QUERY.search,
+		tier = DEFAULT_THEME_QUERY.tier,
+	} = query;
+
+	const recordThemeClick = ( event, { action = '', styleVariationSlug } = {} ) => {
+		const adjustedPosition = themePosition + 1;
+
+		const collectionProps =
+			'' !== collectionId
+				? {
+						collection: collectionId,
+						collection_index: ( collectionPosition || 0 ) + 1,
+						is_collection: true,
```

Comment:
Could we also add an `is_collection_view` prop? Maybe as a follow up.

Reply:
Yeah, the hook should be easy to read and extend at will without worries and concerns.

I tried to not change any behaviours to make it easy to compare with what we currently have — hopefully I succeeded?